invenio_vocabularies/services/generators.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+#
+
+"""Vocabulary generators."""
+
+from invenio_access import any_user, authenticated_user
+from invenio_records_permissions.generators import Generator
+from invenio_search.engine import dsl
+
+
+class AnyUser(Generator):
+    """Allows any user."""
+
+    def needs(self, **kwargs):
+        """Enabling Needs."""
+        return [any_user]
+
+    def query_filter(self, **kwargs):
+        """Match only searchable values in search."""
+        return dsl.Q(
+            "bool",
+            must_not=[dsl.Q("term", tags="non-searchable")],
+        )
+
+
+class Tags(Generator):
+    """Allows any user."""
```

Comment:
Discussed, please add more explanations related to `only_authenticated` and its purpose.